src/core/locator/qgslocator.cpp

```diff
@@ -94,10 +94,15 @@ void QgsLocator::registerFilter( QgsLocatorFilter *filter )
   filter->setParent( this );
 
   // restore settings
-  QgsSettings settings;
-  bool enabled = settings.value( QStringLiteral( "locator_filters/enabled_%1" ).arg( filter->name() ), true, QgsSettings::Section::Gui ).toBool();
-  bool byDefault = settings.value( QStringLiteral( "locator_filters/default_%1" ).arg( filter->name() ), filter->useWithoutPrefix(), QgsSettings::Section::Gui ).toBool();
-  QString prefix = settings.value( QStringLiteral( "locator_filters/prefix_%1" ).arg( filter->name() ), filter->prefix(), QgsSettings::Section::Gui ).toString();
+  bool enabled = true;
+  if ( QgsLocator::Settings::locatorFilterEnabled.exists( filter->name() ) )
```

Comment:
Couldn't we skip all the exists checks here by instead returning the default value when value() is called and the setting doesn't exist?

Reply:
For the enable yes. For the other two i think it is not possible because for example filter->prefix() would return a different value for each subclass of QgsLocatorFilter. QgsSettingsEntry support only one default value passed to the constructor.
Maybe it would be possible to add a second paramete to value( dynamicKey, defaultValueOverride ) or similar. defaultValueOverride would be taken in place of the "normal" default value when is valid.

Reply:
Parameters to override the default value added in commit abb8e80c1f006933eb05969c9fa88be69144c34e